test/jest/interceptor.test.js

```diff
@@ -0,0 +1,207 @@
+'use strict'
+
+const { createServer } = require('http')
+const { Agent, request } = require('../../index')
+/* global expect */
+
+test('interceptors are applied on client from an agent', async () => {
+  const server = createServer((req, res) => {
+    res.setHeader('Content-Type', 'text/plain')
+    res.end('hello')
+  })
+
+  try {
+    const interceptors = []
+    const buildInterceptor = dispatch => {
+      const interceptorContext = { requestCount: 0 }
+      interceptors.push(interceptorContext)
+      return (opts, handler) => {
+        interceptorContext.requestCount++
+        return dispatch(opts, handler)
+      }
+    }
+
+    await new Promise(resolve => server.listen(0, () => resolve()))
+    const opts = { interceptors: { Client: [buildInterceptor] } }
+    const agent = new Agent(opts)
+    const origin = new URL(`http://localhost:${server.address().port}`)
+    await Promise.all([
+      request(origin, { dispatcher: agent }),
+      request(origin, { dispatcher: agent })
+    ])
+
+    // Assert that the requests are run on different interceptors (different Clients)
+    const requestCounts = interceptors.map(x => x.requestCount)
+    expect(requestCounts).toEqual([1, 1])
+  } finally {
```

Comment:
why? can't this be handled in jest `after` block?

Reply:
Because I am not a javascript programmer and I don't use jest. I had enough trouble getting this work...